docs/scheduler.md

```diff
@@ -0,0 +1,152 @@
+
+The purpose this new scheduler is to improve fairness and throughput, to more effectively schedule large and distributed jobs, and to provide a way of running systems with dynamically changing resource requirements (e.g., Spark clusters) via Armada.
+
+## Armada architecture
+
+Armada consists of the Armada server and one or more Kubernetes worker clusters. Each worker cluster consists of a large number of nodes responsible for executing tasks and is managed by a process referred to as the Armada executor. Jobs are submitted to the Armada server, where they are queued until being handed off to an executor, which assigns them to nodes.
+
+Armada uses a two-stage scheduling approach, consisting of a top-level scheduler (which is part of the server), and a set of per-cluster schedulers- one per executor. The top-level scheduler is responsible for assigning jobs to executors and the per-cluster schedulers are responsible for assigning those jobs to nodes. This two-stage approach allows the scheduler to manage very large numbers of nodes. We refer to the top-level scheduler simply as "the scheduler". The job of the scheduler is to decide in what order to assigns jobs to worker clusters.
+
+### Jobs
+
+Each Armada job consists of a bag of Kubernetes object (e.g., pods, services, and ingresses) with a shared life cycle; all objects are created at the start of the job and are destroyed when the job has finished. Each job is a member of a queue (e.g., corresponding to a particular user) and a job set (a per-queue logical grouping of jobs that can be managed as a unit), and has a priority associated with it, i.e., each job has associated with it a tuple `(queue, jobSet, priority)`. Each job is assigned to a cluster as a unit, i.e., all resources that make up a job are created within the same cluster.
+
+Armada jobs are created by submitting a job specification to the Armada server, either via client libraries or via the `armadactl` command-line-utility. In this way, users can submit jobs to Armada directly. However, users may also submit workflows to higher-level schedulers, such as [Apache AirFlow](https://airflow.apache.org/), which in turn submit jobs to Armada.
+
+### Distributed jobs
+
+Distributed jobs are represented as [PodGroups](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/kep/42-podgroup-coscheduling/README.md), i.e., groups of cooperating pods, and the pods that make up a `PodGroup` are scheduled jointly using a gang-scheduling algorithm. Similar to to [kube-batch](https://github.com/kubernetes-sigs/kube-batch) and [Volcano](https://volcano.sh/en/docs/podgroup/), the gang-scheduling algorithm is implemented as a [custom Kubernetes scheduler](https://kubernetes.io/docs/tasks/extend-kubernetes/configure-multiple-schedulers/).
```

Comment:
Do we have anything to say about pre-empting distributed jobs.

Reply:
I added a comment on this.